scripts/setup-adapters.sh

```diff
@@ -16,44 +16,28 @@
 # Propagate errors and improve debugging.
 set -eufx -o pipefail
 
+SCRIPTDIR=$(dirname "$0")
+source $SCRIPTDIR/setup-helper-functions.sh
+DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
+
 function install_aws-sdk-cpp {
-  local NAME="aws-sdk-cpp"
+  local AWS_REPO_NAME="aws/aws-sdk-cpp"
   local AWS_SDK_VERSION="1.9.96"
 
-  if [ -d "${NAME}" ]; then
-    read -r -p "do you want to rebuild '${NAME}'? (y/n) " confirm
-    if [[ "${confirm}" =~ ^[yy]$ ]]; then
-      rm -rf "${NAME}"
-    else
-      return 0
-    fi
-  fi
-
-  git clone --depth 1 --recurse-submodules --branch "${AWS_SDK_VERSION}" https://github.com/aws/aws-sdk-cpp.git "${NAME}"
-  mkdir "${NAME}_build"
-  cd "${NAME}_build" || exit
-  cmake ../"${NAME}" \
-    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
-    -DCMAKE_BUILD_TYPE=Debug \
-    -DBUILD_ONLY:STRING="s3;identity-management" \
-    -DBUILD_SHARED_LIBS:BOOL=OFF \
-    -DMINIMIZE_SIZE:BOOL=ON \
-    -DENABLE_TESTING:BOOL=OFF \
-    -DCMAKE_INSTALL_PREFIX="${DEPENDENCY_DIR}/install" \
-    -GNinja \
-    .
-  local _ret=$?
-  if [ $_ret -ne 0 ] ; then
-     echo "cmake returned with exit code $_ret, aborting!" >&2
-     return $_ret
-  fi
-  ninja
-  ninja install
+  github_checkout $AWS_REPO_NAME $AWS_SDK_VERSION --depth 1 --recurse-submodules
+  cmake_install -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS:BOOL=OFF -DMINIMIZE_SIZE:BOOL=ON -DENABLE_TESTING:BOOL=OF
```

Comment:
can you pass all the original CMake arguments from above? That should resolve the build issues as we will only be building fewer components.

Reply:
Sorry can you clarify, pass this ? `-DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS:BOOL=OFF -DMINIMIZE_SIZE:BOOL=ON -DENABLE_TESTING:BOOL=O` to where? I think they are already being passed down to cmake_install?

Reply:
There is a typo here though: `-DENABLE_TESTING:BOOL=OF`